docs/admin/templates/extending-templates/devcontainers.md

```diff
@@ -1,126 +1,345 @@
+  agent_id = coder_agent.main.id
+  url      = "https://github.com/example/project.git"
+  base_dir = "/home/coder/project"
+}
+
+resource "coder_devcontainer" "project" {
+  count            = data.coder_workspace.me.start_count
+  agent_id         = coder_agent.main.id
+  workspace_folder = "/home/coder/project/${module.git_clone[0].folder_name}"
+  depends_on       = [module.git_clone]
+}
+```
 
-## Enable Dev Containers Integration
+## Dev Container Features
 
-To enable the dev containers integration in your workspace, you must set the
-`CODER_AGENT_DEVCONTAINERS_ENABLE` environment variable to `true` in your
-workspace container:
+Enhance your dev container experience with additional features.
+For more advanced use cases, consult the [advanced dev containers doc](./advanced-dev-containers.md).
+
+### Custom applications
+
+```jsonc
+{
+  "customizations": {
+    "coder": {
+      "apps": {
+        "flask-app": {
+          "command": "python app.py",
+          "icon": "/icon/flask.svg",
+          "subdomain": true,
+          "healthcheck": {
+            "url": "http://localhost:5000/health",
+            "interval": 10,
+            "threshold": 10
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+### Agent naming
+
+Coder names dev container agents in this order:
+
+1. `customizations.coder.agent.name` in `devcontainer.json`
+1. `name` in `devcontainer.json`
+1. Directory name that contains the config
+1. `devcontainer` (default)
+
+### Multiple dev containers
 
 ```terraform
+resource "coder_devcontainer" "frontend" {
+  count            = data.coder_workspace.me.start_count
+  agent_id         = coder_agent.main.id
+  workspace_folder = "/home/coder/frontend"
+}
+
+resource "coder_devcontainer" "backend" {
+  count            = data.coder_workspace.me.start_count
+  agent_id         = coder_agent.main.id
+  workspace_folder = "/home/coder/backend"
+}
+```
+
+## Complete Template Examples
+
+You can test the Coder dev container integration and features with these starter templates.
+
+<details><summary>Docker-based template (privileged)</summary>
+
+This version uses a Docker-in-Docker image, so it works even if the host doesn’t expose a Docker socket.
+
+```terraform
+terraform {
+  required_providers {
+    coder = {
+      source = "coder/coder"
+    }
+    docker = {
+      source = "kreuzwerker/docker"
+    }
+  }
+}
+
+provider "coder" {}
+
+data "coder_workspace"       "me"    {}
+data "coder_workspace_owner" "me"    {}
+
 resource "docker_container" "workspace" {
-  count = data.coder_workspace.me.start_count
-  image = "codercom/oss-dogfood:latest"
+  count      = data.coder_workspace.me.start_count
+  image      = "codercom/enterprise-base:ubuntu"           # includes Coder agent
+  name       = "coder-${data.coder_workspace_owner.me.name}-${lower(data.coder_workspace.me.name)}"
+
+  # share host Docker
+  volumes {
+    host_path      = "/var/run/docker.sock"
+    container_path = "/var/run/docker.sock"
+  }
```

Comment:
This is a very bad idea. All workspaces will try to take over each others' devcontainers.